kolibri_exercise_perseus_plugin/generateKATranslations.js

```diff
@@ -0,0 +1,288 @@
+/*
+ * This utility is designed to download Khan Academy po files from crowdin,
+ * search through them for interface strings that we need in the Perseus renderer,
+ * and then upload them to our Crowdin in order to leverage these translated strings
+ * in our own translations.
+ */
+
+
+/*
+ * Utility functions for generating the paths for locale files, both locally and on crowdin
+ */
+const localeFile = (lang, locale) => {
+  if (locale) {
+    if (locale.length <= 2) {
+      locale = locale.toUpperCase();
+    } else {
+      locale = locale[0].toUpperCase() + locale.slice(1);
+    }
+  }
+  const langCode = lang + (locale ? '_' + locale : '');
+  return `${langCode}/LC_FRONTEND_MESSAGES/exercise_perseus_render_module-messages.json`;
+};
+
+// Format the language code how crowdin expects it
+const crowdinLangCode = (lang, locale) => {
+  return lang + (locale ? '-' + locale.toUpperCase() : '');
+};
+
+// Path to the locale file for a language in the local locale folder
+const localLocaleFile = (lang, locale) => {
+  return './locale/' + localeFile(lang, locale);
+}
+
+// Which kolibri branch to upload translations to on Crowdin
+const kolibriReleaseBranch = 'release-v0.6.x';
+
+// Generate path for Kolibri crowdin translation upload
+const remoteLocaleFile = () => {
+  return `${kolibriReleaseBranch}/` + localeFile('en');
+}
+
+// Load upthe current English translations file, to give us basis for mapping
+// KA translations to Kolibri message ids
+const en = require(localLocaleFile('en'));
+// Po file parser
+const po = require('pofile');
+const fs = require('fs');
+// Our function for getting i18n messages out of the Perseus source code
+const getMessages = require('./getMessages');
+// A function for turning a simple ngettext formatted message to ICU syntax
+const gettextToICU = require('./gettextToICU');
+const https = require('https');
+const AdmZip = require('adm-zip');
+// Load in API keys from the secrets file.
+const {
+  kaApiKey,
+  kolibriApiKey,
+} = require('../crowdinSecrets');
+const request = require('request-promise');
+const temp = require('temp').track();
+const ProgressBar = require('progress');
+
+// List of currently supported languages and associated language and locale codes.
+const languages = {
```

Comment:
Indeed - also would be good to have a centralized list of 'available languages' for Kolibri.